symphony-bdk-core-invokers/symphony-bdk-core-invoker-api/src/main/java/com/symphony/bdk/core/api/invoker/util/PaginatedService.java

```diff
@@ -0,0 +1,124 @@
+package com.symphony.bdk.core.api.invoker.util;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Retrieve a list of elements given a {@link PaginatedApi}.
+ * Goal is to fetch elements lazily.
+ * @param <T> the type of objects to retrieve
+ */
+public class PaginatedService<T> {
```

Comment:
should we keep it internal?

Reply:
Also, why this class goes to the `invoker` module?